Code/GraphMol/MarvinParse/MarvinDefs.cpp

```diff
@@ -63,11 +53,570 @@ ptree MarvinMolBase::toPtree() const {
+            v.second.get<std::string>("<xmlattr>.elementType", "");
+
+        if (mrvAtom->id == "" || mrvAtom->elementType == "") {
+          throw FileParseException(
+              "Expected id, elementType for an atom definition in MRV file");
+        }
+
+        std::string x2 = v.second.get<std::string>("<xmlattr>.x2", "");
+        std::string y2 = v.second.get<std::string>("<xmlattr>.y2", "");
+
+        // x2 and y2 are doubles
+
+        if (x2 != "" && y2 != "" &&
+            (!getCleanDouble(x2, mrvAtom->x2) ||
+             !getCleanDouble(y2, mrvAtom->y2))) {
+          throw FileParseException(
+              "The values x2 and y2 must be large floats in MRV file");
+        }
+
+        std::string formalCharge =
+            v.second.get<std::string>("<xmlattr>.formalCharge", "");
+        if (formalCharge != "") {
+          if (!getCleanInt(formalCharge, mrvAtom->formalCharge)) {
+            throw FileParseException(
+                "The value for formalCharge must be an integer in MRV file");
+          }
+        } else {
+          mrvAtom->formalCharge = 0;
+        }
+
+        mrvAtom->radical = v.second.get<std::string>("<xmlattr>.radical", "");
+        if (mrvAtom->radical != "") {
+          if (!boost::algorithm::contains(
+                  marvinRadicalVals,
+                  std::vector<std::string>{mrvAtom->radical})) {
+            std::ostringstream err;
+            err << "The value for radical must be one of "
+                << boost::algorithm::join(marvinRadicalVals, ", ")
+                << " in MRV file";
+            throw FileParseException(err.str());
+          }
+        } else {
+          mrvAtom->radical = "";
+        }
+
+        std::string isotopeStr =
+            v.second.get<std::string>("<xmlattr>.isotope", "");
+        if (isotopeStr != "") {
+          if (!getCleanInt(isotopeStr, mrvAtom->isotope) ||
+              mrvAtom->isotope <= 0) {
+            std::ostringstream err;
+            err << "The value for isotope must be a positive number in MRV file";
+            throw FileParseException(err.str());
+          }
+        } else {
+          mrvAtom->isotope = 0;
+        }
+
+        std::string valenceStr =
+            v.second.get<std::string>("<xmlattr>.mrvValence", "");
+        if (valenceStr != "") {
+          if (!getCleanInt(valenceStr, mrvAtom->mrvValence) ||
+              mrvAtom->mrvValence < 0) {
+            std::ostringstream err;
+            err << "The value for mrvValence must be a positive number in MRV file";
+            throw FileParseException(err.str());
+          }
+        } else {
+          mrvAtom->mrvValence = -1;
+        }
+
+        std::string hCountStr =
+            v.second.get<std::string>("<xmlattr>.hydrogenCount", "");
+        if (hCountStr != "") {
+          if (!getCleanInt(hCountStr, mrvAtom->hydrogenCount) ||
+              mrvAtom->hydrogenCount < 0) {
+            std::ostringstream err;
```

Comment:
See above

Reply:
There are several places where the error returned has information in it.   For those that did not, I have now just used the static string